saleor/warehouse/availability.py

```diff
@@ -99,3 +129,75 @@ def check_stock_quantity_bulk(
 
     if insufficient_stocks:
         raise InsufficientStock(insufficient_stocks)
+
+
+def get_available_quantity(
+    variant: "ProductVariant",
+    country_code: str,
+    channel_slug: str,
+    checkout_lines: Optional[List["CheckoutLine"]] = None,
+    check_reservations: bool = False,
+) -> int:
+    """Return available quantity for given product in given country."""
+    stocks = Stock.objects.get_variant_stocks_for_country(
+        country_code, channel_slug, variant
+    )
+    if not stocks:
+        return 0
+    return _get_available_quantity(stocks, checkout_lines, check_reservations)
+
+
+def is_product_in_stock(
+    product: "Product", country_code: str, channel_slug: str
+) -> bool:
+    """Check if there is any variant of given product available in given country."""
+    stocks = Stock.objects.get_product_stocks_for_country_and_channel(
+        country_code, channel_slug, product
+    ).annotate_available_quantity()
+    return any(stocks.values_list("available_quantity", flat=True))
+
+
+def get_reserved_quantity(
+    stocks: StockQuerySet, lines: Optional[List["CheckoutLine"]] = None
+) -> int:
+    result = (
+        Reservation.objects.filter(
+            stock__in=stocks,
+        )
+        .not_expired()
+        .exclude_checkout_lines(lines)
+        .aggregate(
+            quantity_reserved=Coalesce(Sum("quantity_reserved"), 0),
+        )
+    )  # type: ignore
+
+    return result["quantity_reserved"]
+
+
+def get_reserved_quantity_bulk(
+    stocks: Iterable[Stock],
+    checkout_lines: Iterable["CheckoutLineInfo"],
```

Comment:
Cosmetics - is it possible to have the same interface to both `get_reserved_quantity*` functions? Now one accepts `[CheckoutLineInfo]` and the other `[CheckoutLine]`. Both could be unified to `[CheckoutLine]` and a queryset as the first param. But if it requires to much effort, let's not bother.